platforms/documentation/docs/src/docs/userguide/authoring-builds/build_environment.adoc

```diff
@@ -248,6 +276,65 @@ $ ./gradlew build -Dorg.gradle.caching.debug=false
 org.gradle.caching.debug=false
 ----
 
+=== Gradle properties should not be used in build logic.
+
+[CAUTION]
+====
+
+Gradle properties should not be retrieved using the `Project` object or the `ProviderFactory` class due to the following behavior.
+If a `build.gradle.kts` file contains the following task:
+
+[source,kotlin]
+.build.gradle.kts
+----
+tasks.register("gradlePropertyTest") {
+    val isCI = providers.gradleProperty("org.gradle.caching.debug")
+    doLast {
+        println(isCI.get())
+    }
+}
+----
+
+And the `gradle.properties` file contains the following line:
+
+[source,properties]
+.gradle.properties
+----
+org.gradle.console=verbose
+----
+
+Then running the `gradlePropertyTest` task yields the following results:
+
+[source,text]
+----
+$ ./gradlew gradlePropertyTest
+
+> Task :buildSrc:generateExternalPluginSpecBuilders UP-TO-DATE
+> Task :buildSrc:extractPrecompiledScriptPluginPlugins UP-TO-DATE
+> Task :buildSrc:compilePluginsBlocks UP-TO-DATE
+[...]
+> Task :buildSrc:classes UP-TO-DATE
+> Task :buildSrc:jar UP-TO-DATE
+> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
+> Task :model:gradlePropertyTest
+verbose
+----
+
+The console output is verbose and therefore correct.
+Let's see what happens when we run the same task with the added command line option (note that the `gradle.properties` file has not changed):
+
+[source,text]
+----
+$ ./gradlew :gradlePropertyTest -Dorg.gradle.console=auto
+
+> Task :model:gradlePropertyTest
+verbose
+----
+
+As expected, the console output is short.
+However, the output of `providers.gradleProperty("org.gradle.caching.debug")` is incorrectly `verbose`.
```

Comment:
I honestly don't follow this example at all. 😅 It seems like it's describing a very specific edge case.

If not referencing Gradle properties in build logic is our stance, I would just leave it at that and not go into a length example of why. If there are better APIs to use instead then this would be a good place to mention them. For example, the recently added [BuildFeatures](https://docs.gradle.org/current/javadoc/org/gradle/api/configuration/BuildFeatures.html) API. I'm not sure what our stance is on using [`Gradle#getStartParameter()`](https://docs.gradle.org/current/javadoc/org/gradle/api/invocation/Gradle.html#getStartParameter()), but that appears to be a way to get the value of many Gradle properties.

Reply:
Did the 180 thing there as well. Removing it.